Bluetooth: Shell: Add connect-name command

Thalley opened the pull request:
Add a new command, connect-name, which takes a string
as the single input and uses that to setup a scan filter
and connect to the first device that matches the name.

This command is thus basically shorthand for
bt scan-filter-set <name>
bt scan on
bt scan off
bt connect
bt scan-filter-clear-name

The idea of this command is to not only make it easier
find and connect to devices in a crowded environment,
but also to more easily connect to devices that 
changes their advertised address.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Shell: Add connect-name command

Add a new command, connect-name, which takes a string
as the single input and uses that to setup a scan filter
and connect to the first device that matches the name.

This command is thus basically shorthand for
bt scan-filter-set <name>
bt scan on
bt scan off
bt connect
bt scan-filter-clear-name

The idea of this command is to not only make it easier
find and connect to devices in a crowded environment,
but also to more easily connect to devices that changes
their advertised address.

The implementation is minimal as it uses existing
shell commands.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley commented:
The main motive behind this PR is to make UPF testing easier :)

Thalley commented on subsys/bluetooth/shell/bt.c:
```diff
@@ -206,12 +212,28 @@ static void scan_recv(const struct bt_le_scan_recv_info *info,
 	auto_connect.addr_set = true;
 	bt_addr_le_copy(&auto_connect.addr, info->addr);
-#endif
+
+	/* Use the above auto_connect.addr address to automatically connect */
+	if (auto_connect.connect_name) {
+		cmd_scan_off(ctx_shell);
+
+		/* "connect" is what would be in argv[0] normally */
+		cmd_connect_le(ctx_shell, 1, (char *[]){ "connect" });
+	}
+#endif /* CONFIG_BT_CENTRAL */
```
Good suggestion, thanks

sjanc reviewed (dismissed):
LGTM but I'd prefer to not suggest in commit message that privacy and fixed adv data (name here) is something that should be used :-)

Thalley commented:
> LGTM but I'd prefer to not suggest in commit message that privacy and fixed adv data (name here) is something that should be used :-)

Removed the privacy reference :)
